### PR TITLE
MINOR: Log warn message with details when there's kerberos login issue

### PR DESCRIPTION
**Pull request by cnZach**
…though we will still retry

Currently, we just capture the exception and retry later, we can't figure out what happened at runtime, it would be helpful to log a warn message with details.

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*
Minor logging change, so no extra test is required. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cnZach** [comment]
Hi @omkreddy or @rajinisivaram , what do you think about this? I just want to print out the error. It will be helpful for troubleshooting kerberos login issue.

File: clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java [hunk 1]
```diff
@@ -237,6 +238,7 @@ public LoginContext login() throws LoginException {
                             break;
                         } catch (LoginException le) {
                             if (retry > 0) {
+                                log.warn("[Principal={}]: Error when trying to re-Login, but will retry ", principal, e);
```
Comment (omkreddy) [inline clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java]: e => le ?

Comment (cnZach) [reply clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java]: oops, yes, corrected.

**Comment by omkreddy** [comment]
ok to test

**Comment by cnZach** [comment]
Not sure what happened... Can you trigger the test again?

**Comment by omkreddy** [comment]
@cnZach  can you rebase this PR against trunk? That will trigger the build.

**Comment by cnZach** [comment]
> @cnZach can you rebase this PR against trunk? That will trigger the build.

Hi @omkreddy , do I need to change anything for this PR?

File: clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java [hunk 2]
```diff
@@ -213,6 +213,7 @@ public LoginContext login() throws LoginException {
                             break;
                         } catch (Exception e) {
                             if (retry > 0) {
+                                log.warn("[Principal={}]: Error when trying to renew with TicketCache, but will retry ", principal, e);
```
Comment (rajinisivaram) [inline clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java:216]: @cnZach Looks like the intention was to print out the exception with stack trace? You would want to use:
```
public void warn(String msg, Throwable t);
```

Comment (cnZach) [reply clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java:216]: Hi @rajinisivaram , do you mean we should use  `log.warn(s"[Principal=$principal]: Error when trying to renew with TicketCache, but will retry ", e); ` ?

**Review by omkreddy — APPROVED** [review]
@cnZach Thanks for the PR. LGTM.

File: clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java (hunk 2 above)
Comment (omkreddy) [reply clients/src/main/java/org/apache/kafka/common/security/kerberos/KerberosLogin.java:216]: slf4j logger methods logs exception if the last argument is throwable. so this is fine.

**Pull request merged**